### DISCUSSION TITLE
Use ALBUMARTIST not ARTIST?

### DISCUSSION THREAD

**Discussion by GlowTube** [gerbera/gerbera#1812] [Q&A] [answered]
I've used Mediatomb for years and it still meets my needs, with only one complaint: When displaying audio files via the Artists category, the results use the ARTIST metadata field. As a result, any track that has several artists listed in this field gets its own entry in the Artists list. So, for example, Alan Walker's album "Different World" has TEN entries in the Artists list: one for Alan Walker by himself, and nine for Alan Walker along with the collaborators on each track. If the Artists display used ALBUMARTIST instead, then this recording would have only a single entry (Alan Walker) which would be more useful to me. Can I convince Mediatomb to do that? Does Gerbera do that? Thanks!

**Comment by KarlStraussberger** [comment]
The logic is pretty much the same. You may add additional entries but the "real" support for multi-value properties is still not implemented.

**Comment by ghlupe2** [comment, marked as answer]
No problem for both MT and Gerbera if you have "scripting" activated during compilation (-DWITH_JS=1). I came the same way and strongly recommend changing to Gerbera. 

In Gerbera you would have to copy the `addAudioStructured` code block from common.js to a new file, e.g. `custom.js` and do some slight changes:
```
[...]
    // first gather data
    var title = obj.meta[M_TITLE];
    if (!title) title = obj.title;

    var artist = obj.meta[M_ARTIST];
    if (!artist) {
        artist = 'Unknown';
        artist_full = null;
    } else {
        artist_full = artist;
        desc = artist;
    }

	// get albumartist tag during import ////////
    var albumartist = obj.meta[M_ALBUMARTIST];
    if (!albumartist) {
        albumartist = artist;
    }
	/////////////////////////////////////////////
	
    var album = obj.meta[M_ALBUM];
    if (!album) {
        album = 'Unknown';
    } else {
        desc = desc + ', ' + album;
    }

    if (desc) {
        desc = desc + ', ';
    }

    desc = desc + title;
[...]

    // Album

    // Extra code for correct display of albums with various artists (usually collections)
    var tracktitle = track + title;
   //var album_artist = album + ' - ' + artist;
    var album_artist = album + ' - ' + albumartist;
    if (description) {
        if (description.toUpperCase() === 'VARIOUS') {
            album_artist = album + ' - Various';
            tracktitle = tracktitle + ' - ' + artist;
        }
    }
[...]
```
(see https://docs.gerbera.io/en/stable/scripting.html for details)

In config.xml change 
`<virtual-layout type="builtin" audio-layout="Default">`
to 
`<virtual-layout type="js" audio-layout="Structured">`
and add below a new line:
`<custom-script>/your/path/to/custom.js</custom-script>`
as described here: https://docs.gerbera.io/en/stable/config-import.html#scripting

This is how my album view looked before
![common](https://user-images.githubusercontent.com/16803676/132179014-dc7b0ccf-d9c5-445d-a659-9318ac504c62.png)

and after the changes
![custom](https://user-images.githubusercontent.com/16803676/132179212-bc0cb806-9084-401a-a526-7e78807e73da.png)

**Reply by GlowTube** [reply]
Many thanks for the detailed reply ghlupe2. You've done exactly what I want to do. I'll switch over to Gerbera, follow your recipe, and eliminate all those extraneous artist entries for albums featuring heavy collaborations. Cheers!
